clar2wasm/tests/wasm-generation/blockinfo.rs

```diff
@@ -0,0 +1,69 @@
+use clar2wasm::tools::{crosscheck_compare_only, TestEnvironment};
+use proptest::proptest;
+
+use crate::{block_range, uint};
+
+const BLOCK_INFO: [&str; 8] = [
+    "burnchain-header-hash",
+    "id-header-hash",
+    "header-hash",
+    "miner-address",
+    "block-reward",
+    "miner-spend-total",
+    "miner-spend-winner",
+    "time",
+];
+const BURN_BLOCK_INFO: [&str; 2] = ["header-hash", "pox-addrs"];
+const STACKS_BLOCK_HEIGHT_LIMIT: u32 = 100;
+const BURN_BLOCK_HEIGHT_LIMIT: u32 = 100;
+
+proptest! {
+    #[test]
+    fn crossprop_blockinfo_within_controlled_range(block_height in block_range(STACKS_BLOCK_HEIGHT_LIMIT)) {
+        let mut env = TestEnvironment::default();
+        env.advance_chain_tip(STACKS_BLOCK_HEIGHT_LIMIT);
```

Comment:
This isn't doing anything because crosscheck isn't using this env.

Reply:
I think we need to add a new variant of crosscheck that takes in an env.

Reply:
I've implemented a function (`crosscheck_compare_advancing_tip_by`) where the chain tip is advanced. Also implemented new functions to handle that case in a proper test environment.